public/app/plugins/datasource/elasticsearch/types.ts

```diff
@@ -0,0 +1,25 @@
+import { DataQuery, DataSourceJsonData } from '@grafana/ui';
+
+export interface ElasticsearchOptions extends DataSourceJsonData {
+  index: string;
+  timeField: string;
+  esVersion: number;
+  interval: string;
+  timeInterval: string;
+  maxConcurrentShardRequests?: number;
+  logMessageField?: string;
+  logLevelField?: string;
+}
+
+export interface ElasticsearchAggregation {
+  id: string;
+  type: string;
+  settings?: any;
+}
+
+export interface ElasticsearchQuery extends DataQuery {
```

Comment:
This should be used instead of the `ElasticsearchQuery` interface defined in `ElasticsearchQueryField.tsx`. Add missing fields from `ElasticsearchQueryField.tsx`.

Reply:
So should the `ElasticsearchQuery` interface defined in `ElasticsearchQueryField.tsx` just be removed? (After adding the missing fields to the interface defined in `types.ts`)

Reply:
Yes, there shouldn't be anyone using that any longer.